core: build fails on Visual Studio 2013 and earlier

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => master ( after e225850cc69b8d0ce08cbce6f8cb91150dbdf88e )
- Operating System / Platform => Windows 7 64bit Professional
- Compiler => Visual Studio 2012, 2013

##### Detailed description

<!-- your description -->

related #10737, #10342 

But rather than that, I figured out that Visual Studio 2012 and 2013 also supports ```int8_t``` and others.
The problem is that the source code has to explicitly include ```cstdint``` before using ```int8_t```.
These lines [1](https://github.com/opencv/opencv/blob/221342fb256164b2fca03b17ac42ca2447fd739f/modules/calib3d/src/rho.cpp#L51) [2](https://github.com/opencv/opencv/blob/221342fb256164b2fca03b17ac42ca2447fd739f/modules/calib3d/src/rho.cpp#L151-L152) will be a good proof to show that old Visual Studio handles fixed width type

I'll write a patch later to fix this.

##### Steps to reproduce

Build recent master on Visual Studio 2012/2013.
We'll get following error message
All of the message are triggered by the missing typedef of ```uint8_t```
```
  c:\opencv\modules\core\src\persistence.hpp(47): error C2065: 'uint8_t' : undeclared identifier (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(47): error C2143: syntax error : missing ')' before 'const' (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(47): error C2059: syntax error : ')' (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(48): error C2365: 'base64::base64_encode' : redefinition; previous definition was 'data variable' (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(49): error C2065: 'uint8_t' : undeclared identifier (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(49): error C2143: syntax error : missing ')' before 'const' (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(49): error C2059: syntax error : ')' (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(50): error C2365: 'base64::base64_decode' : redefinition; previous definition was 'data variable' (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(51): error C2065: 'uint8_t' : undeclared identifier (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(51): error C2143: syntax error : missing ')' before 'const' (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(51): error C2059: syntax error : ')' (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(52): error C2365: 'base64::base64_valid' : redefinition; previous definition was 'data variable' (c:\opencv\modules\core\src\persistence.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(47): error C2065: 'uint8_t' : undeclared identifier (c:\opencv\modules\core\src\persistence_base64.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
  c:\opencv\modules\core\src\persistence.hpp(47): error C2143: syntax error : missing ')' before 'const' (c:\opencv\modules\core\src\persistence_base64.cpp) [c:\opencv\build\modules\core\opencv_core.vcxproj]
```

--- a/modules/core/src/persistence.hpp
+++ b/modules/core/src/persistence.hpp
@@ -12,6 +12,9 @@
 #include <sstream>
 #include <string>
 #include <iterator>
+#ifdef _MSC_VER
+#include <cstdint>
+#endif // _MSC_VER
 
 #define USE_ZLIB 1
 #if USE_ZLIB

